File: contests/2022-2023 ICPC Brazil Subregional Programming Contest/e.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
  int n; cin >> n;
  vector<int> flechas(10e6 + 5, 0);
  int ans = 0;

  while(n--){
    int h; cin >> h;
    if(flechas[h]){
      flechas[h]--;
      flechas[h-1]++;
    }
    else{
      ans++;
      flechas[h-1]++;
    }
  }
  cout << ans << endl;
  return 0;
}

File: contests/2022-2023 ICPC Brazil Subregional Programming Contest/e.py
n = input()
baloes = list(map(int, input().split()))
flechas = [0 for _ in range(10**6 + 5)]

ans = 0
for balao in baloes:
  if flechas[balao] >= 1:
    flechas[balao] -= 1
    flechas[balao-1] += 1
  else:
    ans += 1
    flechas[balao-1] += 1
print(ans)